src/main/objects.c

```diff
@@ -991,16 +991,32 @@ static SEXP inherits3(SEXP x, SEXP what, SEXP which)
     else
 	PROTECT(klass = R_data_class(x, FALSE));
 
-    if(!isString(what))
-	error(_("'what' must be a character vector"));
+    int nprot = 1;
+
+	if (!isString(what))
+	{
+		if (inherits(what, "R7_class"))
+		{
+			SEXP R7_ns = R_FindNamespace(mkString("R7"));
+			SEXP cl = PROTECT(lang2(install("R7_class_name"), what));
+			what = PROTECT(eval(cl, R7_ns));
+			nprot += 2;
+		}
+		if (!isString(what))
+			error(_("'what' must be a character vector or an <R7_class>"));
+	}
+
     int j, nwhat = LENGTH(what);
 
-    if( !isLogical(which) || (LENGTH(which) != 1) )
+	if (!isLogical(which) || (LENGTH(which) != 1))
 	error(_("'which' must be a length 1 logical vector"));
     Rboolean isvec = asLogical(which);
 
-    if(isvec)
+    if (isvec)
+	{
```

Comment:
some weird fiddly bits with the indentation here - i feel like we need to get your RStudio preferences synced with what R core uses? This works for me

<img width="255" alt="Screen Shot 2023-02-12 at 5 11 08 PM" src="https://user-images.githubusercontent.com/19150088/218340316-43ffd832-e69e-4232-9bb8-93d08ea9c667.png">

Reply:
Yes,  the length-8 Tab  has been really standard in R's C code.

Reply:
Thank you, this tip helped a lot!

This is my current understanding of the conventions for indentation in R's C code:

Blocks of code that belong to things like a like an `if`, `else`, `for`, function bodies etc, have one additional indentation level relative to their parent expression, where an "indentation level" corresponds to 4 spaces (occasionally 2 spaces). 

After indentation level of a line is established, strings of 8 spaces are replaced with a single tab character, starting from the left.

E.g., if I set my tab-display-width to 8 spaces and make whitespace characters visible, this is what I see in current master branch `do_AT()`: 
![image](https://user-images.githubusercontent.com/8462255/218579845-7079c61a-a371-4025-9fd4-f47f3dc404ea.png)

Taking most of this body and placing it under an `if(IS_S4_OBJECT(object))` (as was recommended in the meeting earlier today), I arrive at this:

![image](https://user-images.githubusercontent.com/8462255/218580249-c9377eda-9188-4e82-b438-f061a4ab7c7c.png)

Reply:
There is a short (possible old) description near the end of the [R Coding Standards](https://rstudio.github.io/r-manuals/r-ints/R-coding-standards.html) section of the R Internals manual.  This also has the ESS config snippet some of us have used for decades :)